Dropped outdated arcfour cipher

dasm opened the pull request:
OpenSSL 7.6 removed support for arcfour. rsync which uses ssh is trying
to establish connection with use of arcfour. For newer installations it
is ending with an error to establish connection.

Removed hardcoded arcfour cipher from code.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/pytest-dev/pytest-cloud/12)
<!-- Reviewable:end -->

dasm committed:
Dropped outdated arcfour cipher

OpenSSL 7.6 removed support for arcfour. rsync which uses ssh is trying
to establish connection with use of arcfour. For newer installations it
is ending with an error to establish connection.

Removed hardcoded arcfour cipher from code.

dasm commented:
Identified the issue with outdated `arcfour` cipher. https://github.com/pytest-dev/pytest-cloud/issues/11

bubenkoff commented on pytest_cloud/rsync.py:
```diff
@@ -86,7 +86,7 @@ def send(self, raises=True):
                     '--inplace '
                     '--delete-excluded '
                     '--delete '
-                    '-e \"ssh -T -c arcfour -o Compression=no -x\" '
```
as I remember, this cipher was passed to have the least possible network overhead, so what is the next another cipher we can pass from the supported ones?

dasm replied:
OK, that makes sense. Let me look around to see which cipher should replace `arcfour`.

dasm replied:
Based on this old post: https://blog.famzah.net/2010/06/11/openssh-ciphers-performance-benchmark/ indeed the best was `arcfour`. But later, author updated his benchmark https://blog.famzah.net/2015/06/26/openssh-ciphers-performance-benchmark-update-2015/ and found that the fastest is `aes128-gcm@openssh.com`.

According to analysis, everything depends on type of platform. AES instruction set, introduced by Intel in 2010 and AMD in mid-2013 is significantly improving speeds of operations.

Do you think we should force ssh  to use this cipher? Would it make sense to leave decision about cipher up to ssh?

bubenkoff replied:
let's use aes128-gcm@openssh.com then
the point of pytest-cloud that it's an opinionated approach, so the less to configure for the user is better

dasm replied:
Replaced with `aes128-gcm@openssh.com`. Thanks

spinus replied:
can't this option be just confuguration with sane default?

bubenkoff replied:
makes sense

dasm replied:
OK, I'll try making it config option.

dasm replied:
@bubenkoff @spinus is it looking better now?

dasm committed:
Replaced outdated cipher selection

OpenSSH 7.6 is dropping support for arcfour. Many versions have already
disabled this cipher as unsafe.

Replaced arcfour with aes128-gcm@openssh.com to allow it to work.

dasm committed:
Make cipher configurable

Default option chosen for ssh cipher is aes128. However, user can
overwrite this by providing --cloud-rsync-cipher argument

bubenkoff reviewed (changes requested):
please add the docs about new option, also please add a changelog entry

dasm committed:
Added new argument to README

bubenkoff commented on README.rst:
```diff
@@ -85,6 +85,8 @@ Command-line options
     for all test nodes.
 
+* `--cloud-rsync-cipher`
+    Optional ssh cipher selection for `rsync` processes. aes128-gcm@openssh.com by default.
```
please add a reason for choosing the non-default cipher

dasm replied:
Added more verbose description.

dasm committed:
Added more verbose description of chosen cipher

bubenkoff commented:
@dasm what is your pypi username, I will add you as a maintainer so that you can make releases yourself

bubenkoff reviewed (changes requested):
@dasm sorry, one more thing, please also add yourself to contributors

dasm committed:
Added myself to AUTHORS

dasm commented:
@bubenkoff my pypi is the same (just created :P). Added myself to AUTHORS too.

bubenkoff commented:
@dasm please don't forget to make and push git tags when making the releases

bubenkoff commented:
and of course, the version has to be changed

dasm commented:
OK. Will change version and push tags. Thanks @bubenkoff

dasm commented:
@bubenkoff I tried to publish new tag on repository, but it's giving me: `failed to push some refs`
I believe I don't have permissions to do so. For PyPI, `Manage` button is greyed out too.

Am I missing something?

bubenkoff commented:
what's the git command you're running?
as for pypi, you don't have to manage it via the web interface, just do: 

```
 python setup.py sdist upload
```

from the project folder

dasm commented:
I successfully uploaded new version to pypi with CLI. I also pushed new tag. Forgot to create tag with `git tag 4.0.0 master` at first place. Doh.

Thank you @bubenkoff
